domain/src/main/java/me/androidbox/domain/authentication/preference/PreferenceRepository.kt

```diff
@@ -0,0 +1,11 @@
+package me.androidbox.domain.authentication.preference
+
+import me.androidbox.domain.authentication.model.LoginUser
+
+interface PreferenceRepository {
+    fun saveToken(token: String)
+    fun saveUserId(userId: String)
+    fun saveFullName(fullName: String)
+    fun retrieveCurrentUserOrNull(): LoginUser?
```

Comment:
I'm not a fan of adding a "OrNull" to a method
Usually, when the return is Nullable like in this case "LoginUser?" it implies that if no user is found, it will return null

Reply:
@MKiperszmid I read a chapter in this book `Effective Kotlin, Best Practices` by Marcin Moskala. That is where I started to follow this pattern. What does @philipplackner think about this naming convention?

Reply:
I don't have a strong opinion here. The internal Kotlin functions also partly use this pattern, for example when accessing a list at an index:
`list.getOrNull(0)`

I personally go with what @MKiperszmid said, but both are obvious to me and not wrong